File: 42587.cpp
// 프린터

#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;

    queue<pair<int, int> > qu;
    priority_queue<int> pri;        // 비교용

    for (int i = 0; i < priorities.size(); i++) {
        qu.push({i, priorities[i]});
        pri.push(priorities[i]);
    }

    while (!qu.empty()) {
        if (qu.front().second == pri.top()) {
            if (qu.front().first == location) {
                return answer + 1;
            } else {
                answer++;
                qu.pop();
                pri.pop();
            }
        } else {
            qu.push(qu.front());
            qu.pop();
        }
    }

    return answer;
}

File: 42587.py
# 프린터
# 2021-09-21

def solution(priorities, location):
    wait, answer = [], []
    for idx, pr in enumerate(priorities):
        wait.append((idx, pr))

    while wait:
        idx, pr = wait.pop(0)

        if pr < max(priorities):
            priorities.append(pr)
            wait.append((idx, pr))
        else:
            answer.append(idx)
        priorities.pop(0)

    return answer.index(location) + 1

print(solution([1, 1, 9, 1, 1, 1], 0))